database-postgres/dbHelpers.js

```diff
@@ -75,12 +75,60 @@ exports.updateReview = (id, data) => {
   const query = `
     UPDATE reviews
     SET ${updates}
-    WHERE reviewId = ${id}
+    WHERE "reviewId" = ${id}
   `
   return client.query(query);
 }
 
 exports.deleteReview = (id) => {
-  const query = `DELETE FROM reviews WHERE reviewId = ${id}`
+  const query = `DELETE FROM reviews WHERE "reviewId" = ${id}`
   return client.query(query);
+}
+
+exports.getReviewSummary = async (id) => {
+  const query = `SELECT rating FROM reviews WHERE "productId" = ${id}`
+  const { rows } = await client.query(query);
+  return {
+    averageRating: rows.length ? (rows.reduce((acc, cur) => acc + cur.rating, 0) / rows.length).toFixed(2) : 0,
+    totalRatings: rows.length,
+    fiveStar: (rows.filter(row => row.rating === 5).length / rows.length * 100).toFixed(2).concat('%'),
+    fourStar: (rows.filter(row => row.rating === 4).length / rows.length * 100).toFixed(2).concat('%'),
+    threeStar: (rows.filter(row => row.rating === 3).length / rows.length * 100).toFixed(2).concat('%'),
+    twoStar: (rows.filter(row => row.rating === 2).length / rows.length * 100).toFixed(2).concat('%'),
+    oneStar: (rows.filter(row => row.rating === 1).length / rows.length * 100).toFixed(2).concat('%')
+  };
+}
+
+exports.getReviewExcerpts = async (id) => {
+  const query = `
+    SELECT "isHelpfulCount", "description"
+    FROM reviews
+    WHERE "productId" = ${id}
+    ORDER BY "isHelpfulCount" DESC
+  `;
+  const { rows } = await client.query(query);
+  return rows.map(row => row.description.split(' ').slice(0, 2).join(' '));
+}
+
+exports.getReviewsByFeature = async (id) => {
+  const query = `
+    SELECT
+      "easeToUse",
+      "voiceRecognition",
+      "techSupport",
+      "valueForMoney",
+      "qualityOfMaterial",
+      "batteryLife"
+    FROM reviews
+    WHERE "productId" = ${id}
+  `;
+  const { rows } = await client.query(query);
```

Comment:
good use of the doublequotes. when postgres lowercased all my props it took me way too long to realize  what was breaking my client and i had to resolve it with a bunch of aliases.